tests/test_assets.py

```diff
@@ -1376,6 +1450,538 @@ def test_error_message_plurality(self,
+                    'end_date': T('2014-01-10'),
+                },
+
+                # second sid per country
+                {
+                    'symbol': 'C',
+                    'asset_name': 'Asset C',
+                    'start_date': T('2014-01-01'),
+                    'end_date': T('2014-01-05'),
+                },
+                {
+                    'symbol': 'A',  # claiming the unused symbol 'A'
+                    'asset_name': 'Asset A',
+                    'start_date': T('2014-01-06'),
+                    'end_date': T('2014-01-10'),
+                },
+            ] * num_exchanges,
+            index=np.repeat(np.arange(num_exchanges * 2), 2),
+        )
+        metadata['exchange'] = np.repeat(
+            ['EXCHANGE %d' % n for n in range(num_exchanges)],
+            4,
+        )
+        exchanges = pd.DataFrame({
+            'exchange': ['EXCHANGE %d' % n for n in range(num_exchanges)],
+            'country_code': [
+                self.country_code(n) for n in range(num_exchanges)
+            ]
+        })
+        self.write_assets(equities=metadata, exchanges=exchanges)
+        finder = self.asset_finder
+
+        # note: these assertions walk forward in time, starting at assertions
+        # about ownership before the start_date and ending with assertions
+        # after the end_date; new assertions should be inserted in the correct
+        # locations
+
+        # no one held 'A' before 01
+        with self.assertRaises(SymbolNotFound):
+            finder.lookup_symbol('A', T('2013-12-31'))
+        for n in range(num_exchanges):
+            with self.assertRaises(SymbolNotFound):
+                finder.lookup_symbol(
+                    'A',
+                    T('2013-12-31'),
+                    country_code=self.country_code(n),
+                )
+
+        # no one held 'C' before 01
+        with self.assertRaises(SymbolNotFound):
+            finder.lookup_symbol('C', T('2013-12-31'))
+        for n in range(num_exchanges):
+            with self.assertRaises(SymbolNotFound):
+                finder.lookup_symbol(
+                    'C',
+                    T('2013-12-31'),
+                    country_code=self.country_code(n),
+                )
+
+        for asof in pd.date_range('2014-01-01', '2014-01-05', tz='utc'):
+            # from 01 through 05 the first sid on the exchange held 'A'
+            with self.assertRaises(MultipleSymbolsFound):
+                finder.lookup_symbol('A', asof)
+
+            for n in range(num_exchanges):
+                A_result = finder.lookup_symbol(
+                    'A',
+                    asof,
+                    country_code=self.country_code(n),
+                )
+                assert_equal(
+                    A_result,
+                    finder.retrieve_asset(n * 2),
+                    msg=str(asof),
+                )
+                # The symbol and asset_name should always be the last held
```

Comment:
These short comments giving informal reasoning/plain english descriptions of the assertions are super helpful.